From 8854e484c0239e9c79c0add64ea3826eb5a94cf6 Mon Sep 17 00:00:00 2001
From: Liu Yang <yangx.d.liu@intel.com>
Date: Thu, 19 May 2016 19:10:39 +0800
Subject: [PATCH 2/3] Touch:silead:Make the pressure and size value range
 reasonable.

On a Nexus device the pressure maximum value is over 1.5 and the size maximum value is 1.
So we modify the report range comparable to the Nexus.

Change-Id: I2642e26638b9b842004593883a0e43859c1bc11d
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-32032
Signed-off-by: Liu Yang <yangx.d.liu@intel.com>
Reviewed-on: https://android.intel.com:443/507404
---
 drivers/input/touchscreen/silead.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/input/touchscreen/silead.c b/drivers/input/touchscreen/silead.c
index 08d6549..5c60f05 100644
--- a/drivers/input/touchscreen/silead.c
+++ b/drivers/input/touchscreen/silead.c
@@ -84,6 +84,9 @@
 #define SILEAD_PRESSURE		50
 #define SILEAD_FW_NAME_LEN	30
 
+#define PRESSURE_SCALE 6
+#define SIZE_SCALE    4.5
+
 enum silead_ts_power {
 	SILEAD_POWER_ON  = 1,
 	SILEAD_POWER_OFF = 0
@@ -173,8 +176,8 @@ static void silead_ts_report_touch(struct silead_ts_data *data,
 			(swap ? cinfo->y[i] : cinfo->x[i]));
 	input_report_abs(data->input_dev, ABS_MT_POSITION_Y,
 			(swap ? cinfo->x[i] : cinfo->y[i]));
-	input_report_abs(data->input_dev, ABS_MT_PRESSURE, report_pressure[i]);
-	input_report_abs(data->input_dev, ABS_MT_TOUCH_MAJOR, report_pressure[i]);
+	input_report_abs(data->input_dev, ABS_MT_PRESSURE, PRESSURE_SCALE*report_pressure[i]);
+	input_report_abs(data->input_dev, ABS_MT_TOUCH_MAJOR, SIZE_SCALE*report_pressure[i]);
 }
 
 static void silead_ts_set_power(struct i2c_client *client,
-- 
1.9.1

